app/models/state_file_w2.rb

```diff
@@ -15,6 +15,7 @@
 #  local_income_tax_amount     :decimal(12, 2)
 #  local_wages_and_tips_amount :decimal(12, 2)
 #  locality_nm                 :string
+#  state_code                  :string
```

Comment:
[pebble] Why add state code instead of getting it from the intake? And maybe adding a convenience method on this model?

Reply:
could a single intake conceivably have several w2s that are each from different states?

Reply:
It felt weird to me if the validation is on a W2 level, then we should be considering the state code as listed on the W2, not the intake it is attached to. I can confirm with product though

Reply:
Confirmed with Amani that this is a W2-specific validation, not based on the taxpayer's state (ie, intake). Moot point for this filing year but for the logical approach, I think it is correct to keep as is.

Reply:
The W2's StateAbbreviationCd refers to the employer's address. If this change should apply to all W2s for any NJ filer, than we should be determining whether to use it based on the intake's state code, not the W2's.

Reply:
If this is just a NJ filer specific rule and not dependent on NJ employers, we can use the method def validate_state_specific_w2_requirements to make this a state specific rule instead of adding the state code to the model.